contracts/foundry-tests/oracles/PoolOracle.t.sol

```diff
@@ -0,0 +1,385 @@
+            rewind(1 seconds);
+        }
+
+        // Next update is written on the 0 index
+        skip(1 hours);
+        oracle.update(pool);
+        (ts, ratio) = oracle.poolObservations(pool, 0);
+        (, writes) = vm.accesses(address(oracle));
+        assertEq(ts, block.timestamp);
+        assertEq(ratio, cumulativeBalancesRatio);
+        assertEq(writes.length, 2);
+    }
+
+    function testPeek() public {
+        address pool = Mocks.mock("IPool");
+
+        // Saturday, 26 February 2022 10:15:28
+        vm.warp(1645870528);
+        IPool(pool).cumulativeBalancesRatio.mock(6081248333687398745122493757731327);
+        IPool(pool).getCache.mock(1102997721918618810252394, 1199964191067551228067732, 1645859234);
+        oracle.update(pool);
+
+        vm.record();
+
+        // Sunday, 27 February 2022 10:15:27 (24h after)
+        vm.warp(1645956927);
+
+        PoolOracle.Observation memory oldestObservation = oracle.getOldestObservationInWindow(pool);
+        assertEq(oldestObservation.timestamp, 1645870528);
+        // https://www.wolframalpha.com/input?i=6081248333687398745122493757731327+%2B+%28%28%281199964191067551228067732+*+1e27%29+%2F+1102997721918618810252394%29+*+%281645870528+-+1645859234%29%29
+        // 6.09353520908578405938336777296503555592051557672491409350744... × 10^33
+        assertEq(oldestObservation.ratioCumulative, 6093535209085784059383367772965035);
+
+        IPool(pool).cumulativeBalancesRatio.mock(6177238896109883718081380861139109);
+        IPool(pool).getCache.mock(1146659164970519061317333, 1231180360591421143492220, 1645948463);
+
+        // https://www.wolframalpha.com/input?i=6177238896109883718081380861139109+%2B+%28%28%281231180360591421143492220+*+1e27%29+%2F+1146659164970519061317333%29+*+%281645956927+-+1645948463%29%29
+        // 6.18632678455170653998367337614150763172612596547147924230199... × 10^33
+        // https://www.wolframalpha.com/input?i=%286.186326784551706539983673376-6.093535209085784059383367772%29+%2F+%281645956927+-+1645870528%29%29
+        // 1.0739889983208426092929964930149654509890160765749603583374... × 10^-6
+        assertEq(oracle.peek(pool), 1073988998320842609);
+
+        (, bytes32[] memory writes) = vm.accesses(address(oracle));
+        assertEq(writes.length, 0);
+        // Index 6 for 06h (belt and braces)
+        (uint256 ts, uint256 ratio) = oracle.poolObservations(pool, 6);
+        assertEq(ts, 0);
+        assertEq(ratio, 0);
+    }
+
+    function testGet() public {
+        address pool = Mocks.mock("IPool");
+
+        // Saturday, 26 February 2022 10:15:28
+        vm.warp(1645870528);
+        IPool(pool).cumulativeBalancesRatio.mock(6081248333687398745122493757731327);
+        IPool(pool).getCache.mock(1102997721918618810252394, 1199964191067551228067732, 1645859234);
+        oracle.update(pool);
+
+        vm.record();
+
+        // Saturday, 26 February 2022 17:15:28 (6h after)
```

Comment:
same comment as above about adding some comments to help someone reviewing this for the first time.  like explaining a big about what are all these magic numbers?